### PR TITLE
ADT: fix neighbor constness for GenericGraph

### PR DESCRIPTION
**Pull request by AlviseDeFaveri**
Problem:
1. `ForwardNode::getConstNeighbor` returns a pointer to non-const object
2. `FowardNode::toNeighborRange` checks if the type of the argument is
const, but since `Successors` is not declared const the resulting
`toNeighborRange(Successors)` is never executed in the const version

Solution:
1. Make `getConstNeighbor()` return a pointer to a const object
2. Explicitly declare a separate `toNeighborRange()` function which
returns a `const_child_iterator` and remove the template param

### CODE REVIEW

**Comment by aleclearmind** [comment]
Is this @pfez approved?

**Comment by pfez** [comment]
Yes, it's the result of a debugging session I had with @AlviseDeFaveri 

But the commit message is wrong in the parts related to point 1.

**Comment by pfez** [comment]
I would reword the parts related to point 1 in the commit message. Something along these lines:

> Problem:
> 1. `ForwardNode::getConstNeighbor` returns a reference to a non-`const` pointer, but the referenced pointer is `const` 
...

> Solution:
> 1. Make `getConstNeighbor()` return a reference to a `const` pointer

**Comment by pfez** [comment]
The differences are subtle, but the issue is subtle, and I think it's better to try to be precise.

**Comment by pfez** [comment]
Oh, and finally, I think there's a `clang-format` violation here: https://github.com/revng/revng/blob/fe01bcc469c1115714e752f65d744005687c7d53/include/revng/ADT/GenericGraph.h#L155

The leading whitespaces whould be removed.

Besides these two cosmetics changes it's good to merge.

**Review by pfez — APPROVED** [review]
@aleclearmind, I have already expressed a couple of small concerns before you requested my review. @AlviseDeFaveri has already addressed them. To me, it now looks good to merge!

**Comment by aleclearmind** [comment]
Gotcha, will do a final pass soon and then merge.
Thanks.

**Comment by aleclearmind** [comment]
Merged in 9eb9e9b1a121fce7dcec69ad7ff569ee1f1ad39c.
Thanks and congratulations for your first commit in mainline `:)`!
